src/Nethermind/Nethermind.Merge.Plugin/EngineDebugBridge.cs

```diff
@@ -0,0 +1,167 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Blockchain;
+using Nethermind.Blockchain.Blocks;
+using Nethermind.Blockchain.Find;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Blockchain.Tracing;
+using Nethermind.Blockchain.Tracing.GethStyle;
+using Nethermind.Config;
+using Nethermind.Consensus;
+using Nethermind.Consensus.ExecutionRequests;
+using Nethermind.Consensus.Processing;
+using Nethermind.Consensus.Tracing;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.ExecutionRequest;
+using Nethermind.Core.Specs;
+using Nethermind.Crypto;
+using Nethermind.Db;
+using Nethermind.Evm.State;
+using Nethermind.Evm.Tracing;
+using Nethermind.Facade.Eth.RpcTransaction;
+using Nethermind.JsonRpc;
+using Nethermind.JsonRpc.Data;
+using Nethermind.JsonRpc.Modules;
+using Nethermind.JsonRpc.Modules.DebugModule;
+using Nethermind.Merge.Plugin.Data;
+using Nethermind.State;
+using Nethermind.Synchronization.ParallelSync;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nethermind.Merge.Plugin
+{
+    public class EngineDebugBridge : IEngineDebugBridge
+    {
+        private IBlockchainProcessor _blockchainProcessor;
+        private IBlockTree _blockTree;
+        private ISpecProvider _specProvider;
+
+        public EngineDebugBridge(
+            IBlockchainProcessor blockchainProcessor,
+            IBlockTree blockTree,
+            ISpecProvider specProvider)
+        {
+            _blockTree = blockTree ?? throw new ArgumentNullException(nameof(blockTree));
+            _blockchainProcessor = blockchainProcessor ?? throw new ArgumentNullException(nameof(blockchainProcessor));
+            _specProvider = specProvider ?? throw new ArgumentNullException(nameof(specProvider));
+        }
+
+        public ExecutionPayloadForDebugRpc GenerateNewPayload(BlockParameter blockParameter)
+        {
+            SearchResult<Block> searchResult = _blockTree.SearchForBlock(blockParameter);
+            if (searchResult.IsError)
+            {
+                throw new InvalidDataException(searchResult.Error);
```

Comment:
Can we avoid exception and do like:

`public bool TryGenerateNewPayload(BlockParameter blockParameter, out ExecutionPayloadForDebugRpc payload)`